onvif/client.py

```diff
@@ -289,6 +293,37 @@ async def close(self):
         for service in self.services.values():
             await service.close()
 
+    def get_snapshot(self, url, basic_auth):
+        """Get a snapshot image from the camera."""
+        if url is None:
```

Comment:
Since we already have the ONVIF client at this point, we should probably just implement the logic to retrieve the snapshot url here instead of passing it in.

```python
media_service = self.create_media_service()
req = media_service.create_type("GetSnapshotUri")
req.ProfileToken = profile_token # pass this instead of the URL in the method definition.
result = await media_service.GetSnapshotUri(req)
url = result.Uri
```

Reply:
Can implement a cache on the class instance where we do a token/url dictionary lookup instead of retrieving from the device every time.

Reply:
Yeah. I initially had it like this, but the combination of caching (do we need this on the class? I think a single instance is used in HA), and sync vs async, made me drop it.
If we're fine with increasing min python version to 3.7, I'll change this as well.